### PR TITLE
Revert "Referrals entry point (#386)"

### PR DESCRIPTION
**Pull request by ecotopian**
This reverts commit 114f8db00c9bf2d12bffda9158f4a5e4599da4c0.

## Context
- [MOB-1889](https://ecosia.atlassian.net/browse/MOB-1889)
- Relates to https://github.com/ecosia/ios-core/pull/102
- We decided to remove the entry point to referrals from the main menu

### CODE REVIEW

File: Client/Frontend/Browser/MainMenuActionHelper.swift [hunk 1]
```diff
@@ -270,18 +261,6 @@ class MainMenuActionHelper: PhotonActionSheetProtocol, FeatureFlaggable, CanRemo
...
     // MARK: - Actions
 
-    private func getInviteFriendsAction() -> PhotonRowActions {
-        SingleActionViewModel(
-            title: .localized(.inviteFriends),
-            iconString: "inviteFriends",
-            isNew: User.shared.referralsEntryPointIsNew) { [weak self] action in
-                action.isNew = false
-                self?.delegate?.showReferrals()
-                User.shared.referralsEntryPointUsed()
```
Comment (lucaschifino) [inline Client/Frontend/Browser/MainMenuActionHelper.swift:280]: If I looked at it right, I think there is another instance of `User.shared.referralsEntryPointUsed()` inside `MultiplyImpact` that is not covered by this revert 🤔 

Would suggest maybe updating core in here after https://github.com/ecosia/ios-core/pull/102 is merged to be sure any potential issue is covered.

**Review by lucaschifino — COMMENTED** [review]
Looks good! Pointed to a potential remaining un-removed reference that might require and additional look

File: Client/Frontend/Browser/MainMenuActionHelper.swift (hunk 1 above)
Comment (ecotopian) [reply Client/Frontend/Browser/MainMenuActionHelper.swift:280]: Well spotted 🙏 Consider it fixed 😌

**Review by lucaschifino — APPROVED** [review]
Nice 👏 All good to go now 🟢

**Pull request merged**